server/protocols/jmap-integration-testing/jmap-integration-testing-common/src/test/java/org/apache/james/jmap/methods/integration/SetMessagesMethodTest.java

```diff
@@ -5869,4 +5888,137 @@ public void mimeMessageIdShouldBePreservedWhenSending() {
+            "      }}" +
+            "    }," +
+            "    \"#0\"" +
+            "  ]" +
+            "]";
+        given()
+            .header("Authorization", senderAtDestinationDomainToken.serialize())
+            .body(sendMessageBody)
+        .when()
+            .post("/jmap")
+        .then()
+            .statusCode(200)
+            .body(NAME, equalTo("messagesSet"))
+            .body(ARGUMENTS + ".notCreated", aMapWithSize(0))
+            .body(ARGUMENTS + ".created", aMapWithSize(1));
+
+        calmlyAwait
+            .pollDelay(Duration.FIVE_HUNDRED_MILLISECONDS)
+            .atMost(30, TimeUnit.SECONDS)
+            .untilAsserted(() ->
+                assertThat(getLatestMessageId(receiverAtDestinationDomainToken, Role.INBOX))
+                    .isNotNull());
+
+        assertThat(dataProbe.listUsers())
+            .doesNotContain("user@domain2.com");
+        assertThat(mailboxProbe.listUserMailboxes("user@domain2.com"))
+            .isEmpty();
+    }
+
+
+    @Test // MAILET-136
+    public void sendShouldSaveToAsTheAliasAddressWhenDomainAliasMapping() throws Exception {
+        dataProbe.addDomainAliasMapping(ALIAS_DOMAIN, DESTINATION_DOMAIN);
+
+        String messageContent = "content content";
+        String sendMessageBody =
+            "[" +
+            "  [" +
+            "    \"setMessages\"," +
+            "    {" +
+            "      \"create\": { \"creationId1337\" : {" +
+            "        \"from\": { \"name\": \"Sender\", \"email\": \"" + SENDER_AT_DESTINATION_DOMAIN + "\"}," +
+            "        \"to\": [{ \"name\": \"User\", \"email\": \"" + RECEIVER_AT_ALIAS_DOMAIN + "\"}]," +
+            "        \"subject\": \"Thank you for joining example.com!\"," +
+            "        \"textBody\": \"" + messageContent + "\"," +
+            "        \"mailboxIds\": [\"" + getOutboxId(senderAtDestinationDomainToken) + "\"]" +
+            "      }}" +
+            "    }," +
+            "    \"#0\"" +
+            "  ]" +
+            "]";
+        given()
+            .header("Authorization", senderAtDestinationDomainToken.serialize())
+            .body(sendMessageBody)
+        .when()
+            .post("/jmap")
+        .then()
+            .statusCode(200)
+            .body(NAME, equalTo("messagesSet"))
+            .body(ARGUMENTS + ".notCreated", aMapWithSize(0))
+            .body(ARGUMENTS + ".created", aMapWithSize(1));
+
+        calmlyAwait
+            .pollDelay(Duration.FIVE_HUNDRED_MILLISECONDS)
+            .atMost(30, TimeUnit.SECONDS)
+            .untilAsserted(() ->
+                assertThat(getLatestMessageId(senderAtDestinationDomainToken, Role.SENT))
+                    .isNotNull());
+
+        String sentMessageId = getLatestMessageId(senderAtDestinationDomainToken, Role.SENT);
+        assertThat(receiversOfMessage(senderAtDestinationDomainToken, sentMessageId))
+            .containsOnly(RECEIVER_AT_ALIAS_DOMAIN);
+    }
```

Comment:
why not having put these new test cases in a new class, instead of the already overloaded SetMessagesMethodTest?